File: plugins/OrchidPluginDocs/src/main/java/com/eden/orchid/pluginDocs/lists/ComponentsList.java
package com.eden.orchid.pluginDocs.lists;

import com.eden.orchid.api.server.admin.AdminList;
import com.eden.orchid.api.theme.components.OrchidComponent;

import javax.inject.Inject;
import java.util.Collection;
import java.util.Set;
import java.util.TreeSet;

public final class ComponentsList implements AdminList<OrchidComponent> {

    private Set<OrchidComponent> list;

    @Inject
    public ComponentsList(Set<OrchidComponent> list) {
        this.list = new TreeSet<>(list);
    }

    @Override
    public String getKey() {
        return "components";
    }

    @Override
    public Collection<OrchidComponent> getItems() {
        return list;
    }

    @Override
    public String getItemId(OrchidComponent item) {
        return item.getClass().getSimpleName();
    }

    @Override
    public OrchidComponent getItem(String id) {
        for(OrchidComponent item : getItems()) {
            if(id.equals(item.getClass().getSimpleName())) {
                return item;
            }
        }
        return null;
    }
}

File: plugins/OrchidPluginDocs/src/main/kotlin/com/eden/orchid/plugindocs/lists/ComponentsList.kt
package com.eden.orchid.plugindocs.lists

import com.eden.orchid.api.server.admin.AdminList
import com.eden.orchid.api.theme.components.OrchidComponent
import java.util.TreeSet
import javax.inject.Inject

@JvmSuppressWildcards
class ComponentsList @Inject
constructor(list: Set<OrchidComponent>) : AdminList {

    private val list: Set<OrchidComponent>

    init {
        this.list = TreeSet(list)
    }

    override fun getKey(): String {
        return "components"
    }

    override fun getItems(): Collection<Any> {
        return list
    }

    override fun getItemId(item: Any): String {
        return item.javaClass.simpleName
    }

    override fun getItem(id: String): Any? {
        for (item in items) {
            if (id == item.javaClass.simpleName) {
                return item
            }
        }
        return null
    }
}

